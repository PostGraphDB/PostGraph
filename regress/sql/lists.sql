/*
 * Copyright (C) 2023-2024 PostGraphDB
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 *
 * Portions Copyright (c) 2020-2023, Apache Software Foundation
 * Portions Copyright (c) 2019-2020, Bitnine Global
 */ 

LOAD 'postgraph';
SET search_path TO postgraph;

CREATE GRAPH lists;
USE GRAPH lists;

--
-- list literal
--
-- empty list
RETURN [];

-- list of scalar values
RETURN ['str', 1, 1.0, true, null] ;

-- nested lists
RETURN [['str'], [1, [1.0], [[true]]], null] ;

--list concatenation
RETURN ['str', 1, 1.0] + [true, null];

-- IN
RETURN 1 IN ['str', 1, 1.0, true, null];
RETURN 'str' IN ['str', 1, 1.0, true, null];
RETURN 1.0 IN ['str', 1, 1.0, true, null];
RETURN true IN ['str', 1, 1.0, true, null];
RETURN [1,3,5,[2,4,6]] IN ['str', 1, 1.0, true, null, [1,3,5,[2,4,6]]];
RETURN {bool: true, int: 1} IN ['str', 1, 1.0, true, null, {bool: true, int: 1}, [1,3,5,[2,4,6]]];
RETURN null IN ['str', 1, 1.0, true, null];
RETURN null IN ['str', 1, 1.0, true];
RETURN 'str' IN null ;
RETURN 0 IN ['str', 1, 1.0, true, null];
RETURN 1.1 IN ['str', 1, 1.0, true, null];
RETURN 'Str' IN ['str', 1, 1.0, true, null];
RETURN [1,3,5,[2,4,5]] IN ['str', 1, 1.0, true, null, [1,3,5,[2,4,6]]];
RETURN {bool: true, int: 2} IN ['str', 1, 1.0, true, null, {bool: true, int: 1}, [1,3,5,[2,4,6]]];
RETURN null IN 'str' ;
RETURN 'str' IN 'str' ;

-- TODO ALL, ANY/SOME, NONE

-- list access
RETURN [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10][0];
RETURN [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10][5];
RETURN [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10][10];
RETURN [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10][-1];
RETURN [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10][-3];
-- should return null
RETURN [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10][11];

-- list slice
RETURN [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10][0..];
RETURN [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10][..11];
RETURN [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10][0..0];
RETURN [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10][10..10];
RETURN [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10][0..1];
RETURN [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10][9..10];
RETURN [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10][-1..];
RETURN [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10][-1..11];
RETURN [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10][-3..11];
RETURN [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10][-1..10];
RETURN 0[0..1];
RETURN 0[[0]..[1]];
RETURN [0][0..-2147483649];

-- access and slice operators nested
CYPHER WITH [0, 1, [2, 3, 4], 5, [6, 7, 8], 9] as l RETURN l[0] ;
CYPHER WITH [0, 1, [2, 3, 4], 5, [6, 7, 8], 9] as l RETURN l[2] ;
CYPHER WITH [0, 1, [2, 3, 4], 5, [6, 7, 8], 9] as l RETURN l[-1] ;
CYPHER WITH [0, 1, [2, 3, 4], 5, [6, 7, 8], 9] as l RETURN l[2][-2] ;
CYPHER WITH [0, 1, [2, 3, 4], 5, [6, 7, 8], 9] as l RETURN l[2][-2..] ;
CYPHER WITH [0, 1, [2, 3, 4], 5, [6, 7, 8], 9] as l RETURN l[-2..] ;
CYPHER WITH [0, 1, [2, 3, 4], 5, [6, 7, 8], 9] as l RETURN l[-2..][-1..][-1..] ;
CYPHER WITH [0, 1, [2, 3, 4], 5, [6, 7, 8], 9] as l RETURN l[-2..][-1..][0] ;
CYPHER WITH [0, 1, [2, 3, 4], 5, [6, 7, 8], 9] as l RETURN l[-2..][-1..][-1] ;
CYPHER WITH [0, 1, [2, 3, 4], 5, [6, 7, 8], 9] as l RETURN l[-2..][-2..-1] ;
CYPHER WITH [0, 1, [2, 3, 4], 5, [6, 7, 8], 9] as l RETURN l[-4..-2] ;
CYPHER WITH [0, 1, [2, 3, 4], 5, [6, 7, 8], 9] as l RETURN l[-4..-2][-2] ;
CYPHER WITH [0, 1, [2, 3, 4], 5, [6, 7, 8], 9] as l RETURN l[-4..-2][0] ;
CYPHER WITH [0, 1, [2, 3, 4], 5, [6, 7, 8], 9] as l RETURN l[-4..-2][-2][-2..] ;
CYPHER WITH [0, 1, [2, 3, 4], 5, [6, 7, 8], 9] as l RETURN l[-4..-2][-2][-2..][0] ;

-- empty list
CYPHER WITH [0, 1, [2, 3, 4], 5, [6, 7, 8], 9] as l RETURN l[-2..][-1..][-2..-2] ;

-- should return null
CYPHER WITH [0, 1, [2, 3, 4], 5, [6, 7, 8], 9] as l RETURN l[2][3] ;
CYPHER WITH [0, 1, [2, 3, 4], 5, [6, 7, 8], 9] as l RETURN l[-2..][-1..][-2] ;

-- size() of a string
 RETURN size('12345') ;
 RETURN size("1234567890") ;
-- size() of an array
 RETURN size([1, 2, 3, 4, 5]) ;
 RETURN size([]) ;
 RETURN size(null) ;
-- head
 RETURN head([1, 2, 3, 4, 5]) ;
 RETURN head([1]) ;
 RETURN head([]) ;
 RETURN head(null) ;
-- last
 RETURN last([1, 2, 3, 4, 5]) ;
 RETURN last([1]) ;
 RETURN last([]) ;
 RETURN last(null) ;

-- range()
RETURN range(0, 10);
RETURN range(0, 10, 1);
RETURN range(0, -10, -3);
RETURN range(0, 10, 11);
RETURN range(-20, 10, 5);
RETURN range(0, 10, -1);
RETURN range(null, -10, -3);
RETURN range(0, null, -3);
RETURN range(0, -10.0, -3.0);


--
-- Cleanup
--
DROP GRAPH lists CASCADE;

--
-- End of tests
--
