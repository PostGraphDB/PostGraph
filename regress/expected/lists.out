/*
 * PostGraph
 * Copyright (C) 2023 PostGraphDB
 * 
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
LOAD 'postgraph';
SET search_path TO postgraph;
SELECT * FROM create_graph('lists');
NOTICE:  graph "lists" has been created
 create_graph 
--------------
 
(1 row)

--
-- list literal
--
-- empty list
SELECT * FROM cypher('lists', $$RETURN []$$) AS r(c gtype);
 c  
----
 []
(1 row)

-- list of scalar values
SELECT * FROM cypher('lists', $$ RETURN ['str', 1, 1.0, true, null] $$) AS r(c gtype);
              c              
-----------------------------
 ["str", 1, 1.0, true, null]
(1 row)

-- nested lists
SELECT * FROM cypher('lists', $$ RETURN [['str'], [1, [1.0], [[true]]], null] $$) AS r(c gtype);
                   c                   
---------------------------------------
 [["str"], [1, [1.0], [[true]]], null]
(1 row)

--list concatenation
SELECT * FROM cypher('lists', $$RETURN ['str', 1, 1.0] + [true, null]$$) AS r(c gtype);
              c              
-----------------------------
 ["str", 1, 1.0, true, null]
(1 row)

-- IN
SELECT * FROM cypher('lists', $$RETURN 1 IN ['str', 1, 1.0, true, null]$$) AS r(c boolean);
 c 
---
 t
(1 row)

SELECT * FROM cypher('lists', $$RETURN 'str' IN ['str', 1, 1.0, true, null]$$) AS r(c boolean);
 c 
---
 t
(1 row)

SELECT * FROM cypher('lists', $$RETURN 1.0 IN ['str', 1, 1.0, true, null]$$) AS r(c boolean);
 c 
---
 t
(1 row)

SELECT * FROM cypher('lists', $$RETURN true IN ['str', 1, 1.0, true, null]$$) AS r(c boolean);
 c 
---
 t
(1 row)

SELECT * FROM cypher('lists', $$RETURN [1,3,5,[2,4,6]] IN ['str', 1, 1.0, true, null, [1,3,5,[2,4,6]]]$$) AS r(c boolean);
 c 
---
 t
(1 row)

SELECT * FROM cypher('lists', $$RETURN {bool: true, int: 1} IN ['str', 1, 1.0, true, null, {bool: true, int: 1}, [1,3,5,[2,4,6]]]$$) AS r(c boolean);
 c 
---
 t
(1 row)

SELECT * FROM cypher('lists', $$RETURN null IN ['str', 1, 1.0, true, null]$$) AS r(c boolean);
 c 
---
 
(1 row)

SELECT * FROM cypher('lists', $$RETURN null IN ['str', 1, 1.0, true]$$) AS r(c boolean);
 c 
---
 
(1 row)

SELECT * FROM cypher('lists', $$RETURN 'str' IN null $$) AS r(c boolean);
 c 
---
 
(1 row)

SELECT * FROM cypher('lists', $$RETURN 0 IN ['str', 1, 1.0, true, null]$$) AS r(c boolean);
 c 
---
 f
(1 row)

SELECT * FROM cypher('lists', $$RETURN 1.1 IN ['str', 1, 1.0, true, null]$$) AS r(c boolean);
 c 
---
 f
(1 row)

SELECT * FROM cypher('lists', $$RETURN 'Str' IN ['str', 1, 1.0, true, null]$$) AS r(c boolean);
 c 
---
 f
(1 row)

SELECT * FROM cypher('lists', $$RETURN [1,3,5,[2,4,5]] IN ['str', 1, 1.0, true, null, [1,3,5,[2,4,6]]]$$) AS r(c boolean);
 c 
---
 f
(1 row)

SELECT * FROM cypher('lists', $$RETURN {bool: true, int: 2} IN ['str', 1, 1.0, true, null, {bool: true, int: 1}, [1,3,5,[2,4,6]]]$$) AS r(c boolean);
 c 
---
 f
(1 row)

SELECT * FROM cypher('lists', $$RETURN null IN 'str' $$) AS r(c boolean);
ERROR:  object of IN must be a list
SELECT * FROM cypher('lists', $$RETURN 'str' IN 'str' $$) AS r(c boolean);
ERROR:  object of IN must be a list
-- TODO ALL, ANY/SOME, NONE
-- list access
SELECT * FROM cypher('lists', $$RETURN [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10][0]$$) AS r(c gtype);
 c 
---
 0
(1 row)

SELECT * FROM cypher('lists', $$RETURN [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10][5]$$) AS r(c gtype);
 c 
---
 5
(1 row)

SELECT * FROM cypher('lists', $$RETURN [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10][10]$$) AS r(c gtype);
 c  
----
 10
(1 row)

SELECT * FROM cypher('lists', $$RETURN [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10][-1]$$) AS r(c gtype);
 c  
----
 10
(1 row)

SELECT * FROM cypher('lists', $$RETURN [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10][-3]$$) AS r(c gtype);
 c 
---
 8
(1 row)

-- should return null
SELECT * FROM cypher('lists', $$RETURN [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10][11]$$) AS r(c gtype);
 c 
---
 
(1 row)

-- list slice
SELECT * FROM cypher('lists', $$RETURN [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10][0..]$$) AS r(c gtype);
                 c                  
------------------------------------
 [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
(1 row)

SELECT * FROM cypher('lists', $$RETURN [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10][..11]$$) AS r(c gtype);
                 c                  
------------------------------------
 [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
(1 row)

SELECT * FROM cypher('lists', $$RETURN [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10][0..0]$$) AS r(c gtype);
 c  
----
 []
(1 row)

SELECT * FROM cypher('lists', $$RETURN [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10][10..10]$$) AS r(c gtype);
 c  
----
 []
(1 row)

SELECT * FROM cypher('lists', $$RETURN [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10][0..1]$$) AS r(c gtype);
  c  
-----
 [0]
(1 row)

SELECT * FROM cypher('lists', $$RETURN [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10][9..10]$$) AS r(c gtype);
  c  
-----
 [9]
(1 row)

SELECT * FROM cypher('lists', $$RETURN [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10][-1..]$$) AS r(c gtype);
  c   
------
 [10]
(1 row)

SELECT * FROM cypher('lists', $$RETURN [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10][-1..11]$$) AS r(c gtype);
  c   
------
 [10]
(1 row)

SELECT * FROM cypher('lists', $$RETURN [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10][-3..11]$$) AS r(c gtype);
     c      
------------
 [8, 9, 10]
(1 row)

SELECT * FROM cypher('lists', $$RETURN [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10][-1..10]$$) AS r(c gtype);
 c  
----
 []
(1 row)

SELECT * from cypher('lists', $$RETURN 0[0..1]$$) as r(a gtype);
ERROR:  slice must access a list
SELECT * from cypher('lists', $$RETURN 0[[0]..[1]]$$) as r(a gtype);
ERROR:  slice must access a list
SELECT * from cypher('lists', $$RETURN [0][0..-2147483649]$$) as r(a gtype);
 a  
----
 []
(1 row)

-- access and slice operators nested
SELECT * from cypher('lists', $$ WITH [0, 1, [2, 3, 4], 5, [6, 7, 8], 9] as l RETURN l[0] $$) as (results gtype);
 results 
---------
 0
(1 row)

SELECT * from cypher('lists', $$ WITH [0, 1, [2, 3, 4], 5, [6, 7, 8], 9] as l RETURN l[2] $$) as (results gtype);
  results  
-----------
 [2, 3, 4]
(1 row)

SELECT * from cypher('lists', $$ WITH [0, 1, [2, 3, 4], 5, [6, 7, 8], 9] as l RETURN l[-1] $$) as (results gtype);
 results 
---------
 9
(1 row)

SELECT * from cypher('lists', $$ WITH [0, 1, [2, 3, 4], 5, [6, 7, 8], 9] as l RETURN l[2][-2] $$) as (results gtype);
 results 
---------
 3
(1 row)

SELECT * from cypher('lists', $$ WITH [0, 1, [2, 3, 4], 5, [6, 7, 8], 9] as l RETURN l[2][-2..] $$) as (results gtype);
 results 
---------
 [3, 4]
(1 row)

SELECT * from cypher('lists', $$ WITH [0, 1, [2, 3, 4], 5, [6, 7, 8], 9] as l RETURN l[-2..] $$) as (results gtype);
    results     
----------------
 [[6, 7, 8], 9]
(1 row)

SELECT * from cypher('lists', $$ WITH [0, 1, [2, 3, 4], 5, [6, 7, 8], 9] as l RETURN l[-2..][-1..][-1..] $$) as (results gtype);
 results 
---------
 [9]
(1 row)

SELECT * from cypher('lists', $$ WITH [0, 1, [2, 3, 4], 5, [6, 7, 8], 9] as l RETURN l[-2..][-1..][0] $$) as (results gtype);
 results 
---------
 9
(1 row)

SELECT * from cypher('lists', $$ WITH [0, 1, [2, 3, 4], 5, [6, 7, 8], 9] as l RETURN l[-2..][-1..][-1] $$) as (results gtype);
 results 
---------
 9
(1 row)

SELECT * from cypher('lists', $$ WITH [0, 1, [2, 3, 4], 5, [6, 7, 8], 9] as l RETURN l[-2..][-2..-1] $$) as (results gtype);
   results   
-------------
 [[6, 7, 8]]
(1 row)

SELECT * from cypher('lists', $$ WITH [0, 1, [2, 3, 4], 5, [6, 7, 8], 9] as l RETURN l[-4..-2] $$) as (results gtype);
    results     
----------------
 [[2, 3, 4], 5]
(1 row)

SELECT * from cypher('lists', $$ WITH [0, 1, [2, 3, 4], 5, [6, 7, 8], 9] as l RETURN l[-4..-2][-2] $$) as (results gtype);
  results  
-----------
 [2, 3, 4]
(1 row)

SELECT * from cypher('lists', $$ WITH [0, 1, [2, 3, 4], 5, [6, 7, 8], 9] as l RETURN l[-4..-2][0] $$) as (results gtype);
  results  
-----------
 [2, 3, 4]
(1 row)

SELECT * from cypher('lists', $$ WITH [0, 1, [2, 3, 4], 5, [6, 7, 8], 9] as l RETURN l[-4..-2][-2][-2..] $$) as (results gtype);
 results 
---------
 [3, 4]
(1 row)

SELECT * from cypher('lists', $$ WITH [0, 1, [2, 3, 4], 5, [6, 7, 8], 9] as l RETURN l[-4..-2][-2][-2..][0] $$) as (results gtype);
 results 
---------
 3
(1 row)

-- empty list
SELECT * from cypher('lists', $$ WITH [0, 1, [2, 3, 4], 5, [6, 7, 8], 9] as l RETURN l[-2..][-1..][-2..-2] $$) as (results gtype);
 results 
---------
 []
(1 row)

-- should return null
SELECT * from cypher('lists', $$ WITH [0, 1, [2, 3, 4], 5, [6, 7, 8], 9] as l RETURN l[2][3] $$) as (results gtype);
 results 
---------
 
(1 row)

SELECT * from cypher('lists', $$ WITH [0, 1, [2, 3, 4], 5, [6, 7, 8], 9] as l RETURN l[-2..][-1..][-2] $$) as (results gtype);
 results 
---------
 
(1 row)

-- size() of a string
SELECT * FROM cypher('lists', $$ RETURN size('12345') $$) AS (size gtype);
 size 
------
 5
(1 row)

SELECT * FROM cypher('lists', $$ RETURN size("1234567890") $$) AS (size gtype);
 size 
------
 10
(1 row)

-- size() of an array
SELECT * FROM cypher('lists', $$ RETURN size([1, 2, 3, 4, 5]) $$) AS (size gtype);
 size 
------
 5
(1 row)

SELECT * FROM cypher('lists', $$ RETURN size([]) $$) AS (size gtype);
 size 
------
 0
(1 row)

SELECT * FROM cypher('lists', $$ RETURN size(null) $$) AS (size gtype);
 size 
------
 
(1 row)

-- head
SELECT * FROM cypher('lists', $$ RETURN head([1, 2, 3, 4, 5]) $$) AS (head gtype);
 head 
------
 1
(1 row)

SELECT * FROM cypher('lists', $$ RETURN head([1]) $$) AS (head gtype);
 head 
------
 1
(1 row)

SELECT * FROM cypher('lists', $$ RETURN head([]) $$) AS (head gtype);
 head 
------
 
(1 row)

SELECT * FROM cypher('lists', $$ RETURN head(null) $$) AS (head gtype);
 head 
------
 
(1 row)

-- last
SELECT * FROM cypher('lists', $$ RETURN last([1, 2, 3, 4, 5]) $$) AS (last gtype);
 last 
------
 5
(1 row)

SELECT * FROM cypher('lists', $$ RETURN last([1]) $$) AS (last gtype);
 last 
------
 1
(1 row)

SELECT * FROM cypher('lists', $$ RETURN last([]) $$) AS (last gtype);
 last 
------
 
(1 row)

SELECT * FROM cypher('lists', $$ RETURN last(null) $$) AS (last gtype);
 last 
------
 
(1 row)

-- range()
SELECT * from cypher('lists', $$RETURN range(0, 10)$$) as (range gtype);
               range                
------------------------------------
 [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
(1 row)

SELECT * from cypher('lists', $$RETURN range(0, 10, 1)$$) as (range gtype);
               range                
------------------------------------
 [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
(1 row)

SELECT * from cypher('lists', $$RETURN range(0, -10, -3)$$) as (range gtype);
      range      
-----------------
 [0, -3, -6, -9]
(1 row)

SELECT * from cypher('lists', $$RETURN range(0, 10, 11)$$) as (range gtype);
 range 
-------
 [0]
(1 row)

SELECT * from cypher('lists', $$RETURN range(-20, 10, 5)$$) as (range gtype);
             range             
-------------------------------
 [-20, -15, -10, -5, 0, 5, 10]
(1 row)

SELECT * from cypher('lists', $$RETURN range(0, 10, -1)$$) as (range gtype);
 range 
-------
 []
(1 row)

SELECT * from cypher('lists', $$RETURN range(null, -10, -3)$$) as (range gtype);
 range 
-------
 
(1 row)

SELECT * from cypher('lists', $$RETURN range(0, null, -3)$$) as (range gtype);
 range 
-------
 
(1 row)

SELECT * from cypher('lists', $$RETURN range(0, -10.0, -3.0)$$) as (range gtype);
      range      
-----------------
 [0, -3, -6, -9]
(1 row)

--
-- Cleanup
--
SELECT * FROM drop_graph('lists', true);
NOTICE:  drop cascades to 2 other objects
DETAIL:  drop cascades to table lists._ag_label_vertex
drop cascades to table lists._ag_label_edge
NOTICE:  graph "lists" has been dropped
 drop_graph 
------------
 
(1 row)

--
-- End of tests
--
